### ISSUE TITLE
[Proposal] add clickable region in InputDecorator

### ISSUE THREAD

**Issue by cedvdb** [flutter/flutter#104345]
# use case
 
 Achieving this kind of input with a inkWell on the left is pretty damn hard:
 
 
![image](https://user-images.githubusercontent.com/22280985/169688066-ea035357-76bc-4447-b9da-2ed0ed8891b2.png)

This is convoluted to do. You have to measure the height of the input decorator, remove the error height when there is an error, change the height when it's underline or outline border, when there is a label or not and so on. Then you have to wrap the input decorator in a stack, have the clickable region be the correct height and width, but it also has to be aligned correctly, the width also depends on a series of factor from InputDecorator. Then comes the aligments issues with the main textField...

I've tried everything, multiple times nothing is straightforward here.

Simply put to achieve something like that you have to hack your way around InputDecorator, because it's not build to do this kind of things. It makes packages harder to maintain that they need to be. 

# proposed

This is just an idea, I'm sure there are better ways but:

 - add `prefixRegion` & `suffixRegion` to input decorator which would add a inkWell around the prefix / suffix.
 - keep prefix correctly aligned

**Comment by HansMuller** [comment]
Adding a pair of Widget valued properties - prefixRegion and suffixRegion - seems like a reasonable proposal. InputDecorator/InputDecoration are already hopelessly complex and one more pair of properties will not change that. The additional widgets would not affect the overall layout and could be made in _RenderDecoration.
